ocaml/xapi/vgpuops.ml

```diff
@@ -129,7 +129,8 @@ let reserve_free_virtual_function ~__context vm pf =
 let add_vgpus_to_vm ~__context host vm vgpus vgpu_manual_setup =
   (* Only support a maximum of one virtual GPU per VM for now. *)
   match vgpus with
-  | [] -> ()
+  | [] ->
+    (try Db.VM.remove_from_other_config ~__context ~self:vm ~key:Xapi_globs.vgpu_pci with _ -> ());
```

Comment:
No - if you replace one vGPU with another then you still need add_pcis_to_vm to remove it from other config before adding it - otherwise the add_to_other_config will fail.
I think if we move the remove_from_other_config to before the "match vgpus" then it would make sense, and it can be removed from add_pcis_to_vm (although it increases the risk of a race condition if something else modifies this key in a different thread)